SimeonInstaller.ps1

```diff
@@ -2685,43 +2685,57 @@ CRLFOption=CRLFAlways
+            }
+            ContentType = 'application/json'
         }
 
-        $contributorsDisplayName = "[$Project]\Contributors"
-        $contributorsId = $identities.results.identities |? displayName -eq $contributorsDisplayName | Select -ExpandProperty localId
-        $projectId = (Get-AzureDevOpsProjectId -Organization $Organization -Project $Project)
+        $projectId = Get-AzureDevOpsProjectId -Organization $Organization -Project $Project
 
-        Write-Information "Making Contributors admin for project library"
-        Invoke-WithRetry { Invoke-RestMethod -Header $authenicationHeader -Uri "https://dev.azure.com/$Organization/_apis/securityroles/scopes/distributedtask.library/roleassignments/resources/$projectId`$0`?api-version=6.1-preview.1" -Method Put -ContentType "application/json" -Body @"
-        [
+        $identities = irm @restProps "https://dev.azure.com/$Organization/_apis/IdentityPicker/Identities" -Method Post -Body @"
             {
-                "roleName": "Administrator",
-                "userId": "$contributorsId"
+                "query": "Contributors",
+                "identityTypes": [
+                    "group"
+                ],
+                "operationScopes": [
+                    "ims",
+                    "source"
+                ],
+                "options": {
+                    "MinResults": 1,
+                    "MaxResults": 1000
+                },
+                "properties": [
+                    "DisplayName"
+                ]
             }
-        ]
 "@
-        } | Out-Null
+
+        $contributorsDisplayName = "[$Project]\Contributors"
+        $contributorsId = $identities.results.identities | ? displayName -eq $contributorsDisplayName | Select -ExpandProperty localId
+
+        Write-Information "Polling to make Contributors admin for project library"
+        $currentContributorsScopeDisplayName = ''
+        while ($currentContributorsScopeDisplayName -ne 'Administrator') {
+            Write-Information "Trying to make Contributors admin for project library"
+            irm @restProps -Uri "https://dev.azure.com/$Organization/_apis/securityroles/scopes/distributedtask.library/roleassignments/resources/$projectId`$0`?api-version=6.1-preview.1" -Method Put -ContentType "application/json" -Body @"
+              [
+                  {
+                      "roleName": "Administrator",
+                      "userId": "$contributorsId"
+                  }
+              ]
+"@ | Out-Null
+
+            $currentScopesUrl = "https://dev.azure.com/$Organization/_apis/securityroles/scopes/distributedtask.library/roleassignments/resources/$($projectId)`$0"
+            $currentScopes = irm @restProps -Uri $currentScopesUrl -Method Get
+            $currentContributorsScope = $currentScopes.value | where { $_.identity.uniqueName -eq $contributorsDisplayName }
+            $currentContributorsScopeDisplayName = $currentContributorsScope.role.displayName
```

Comment:
probably want to add a sleep before the second loop (but only if there will be a second iteration)